src/components/Quiz.tsx

```diff
@@ -1,30 +1,49 @@
-import { useContext, useState } from 'react';
+import { useContext, useState, useEffect } from 'react';
 import Form from './Form';
 import Result from './Result';
-import { Data, QuestionType } from '../types';
+import { Data, QuestionPoolType } from '../types';
 import { Context } from '../Context';
 
 function Quiz({ questions }: { questions: Data }) {
-  const { quizEnded, setQuizEnded } = useContext(Context);
+  const { questionsPool, setQuestionsPool, quizEnded, setQuizEnded } =
+    useContext(Context);
 
+  const [dataIsReady, setDataIsReady] = useState(false);
   const [questionIndex, setQuestionIndex] = useState(0);
-  const [question, setQuestion] = useState<QuestionType | null>(questions[0]);
+  const [question, setQuestion] = useState<QuestionPoolType | null>();
+
+  useEffect(() => {
+    // Initialize pool of questions
+    const pool = questions.map((question) => ({
+      ...question,
+      remainingAttempts: 3,
+      correctlyAnswered: false,
+    }));
+    setQuestionsPool(pool);
+    setQuestion(pool[0]);
+    setDataIsReady(true);
+  }, [questions]);
+
+  useEffect(() => {
+    if (!dataIsReady) return;
+    nextQuestion();
+  }, [questionsPool]);
```

Comment:
And here

```
  30:6  warning  React Hook useEffect has missing dependencies: 'dataIsReady' and 'nextQuestion'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Please resolve them accordingly.

Reply:
This should have been fixed in this commit: https://github.com/christophblessing/quiz-recap/pull/39/commits/60a87292fa0bddd18edd7f9bffa9f11dbf0ab1c1